doc/updateNotebook.js

```diff
@@ -69,47 +69,52 @@ const writeMarkdown = async (
         output.push(`![Image](${pathModule.basename(imagePath)})`);
         output.push('');
       }
-    }
-    if (download) {
-      const { entries } = download;
-      for (let { base64Data, filename, data, path } of entries) {
-        if (!data && base64Data) {
-          data = new Uint8Array(Base64ArrayBuffer.decode(base64Data));
-        }
-        if (!data) {
-          data = await read(path, { workspace });
-        }
-        const observedPath = `${modulePath}.observed.${filename}`;
-        const expectedPath = `${modulePath}.${filename}`;
-        if (!filename.endsWith('stl')) {
-          // STL output has become unstable; skip for now.
-          try {
-            const observed = new TextDecoder('utf8').decode(data);
-            const expected = readFileSync(expectedPath, 'utf8');
-            if (observed !== expected) {
-              failedExpectations.push(`cp '${observedPath}' '${expectedPath}'`);
-            }
-          } catch (error) {
-            console.log(`EE/3: ${JSON.stringify(error)}`);
-            if (error.code === 'ENOENT') {
-              failedExpectations.push(`cp '${observedPath}' '${expectedPath}'`);
-              failedExpectations.push(`git add '${expectedPath}'`);
-            } else {
-              throw error;
+      const { download } = view;
+      if (download) {
+        const { entries } = download;
+        for (let { base64Data, filename, data, path } of entries) {
+          if (!data && base64Data) {
+            data = new Uint8Array(Base64ArrayBuffer.decode(base64Data));
+          }
+          if (!data) {
+            data = await read(path, { workspace });
+          }
+          const observedPath = `${modulePath}.observed.${filename}`;
+          const expectedPath = `${modulePath}.${filename}`;
+          if (!filename.endsWith('stl')) {
+            // STL output has become unstable; skip for now.
+            try {
+              const observed = new TextDecoder('utf8').decode(data);
+              const expected = readFileSync(expectedPath, 'utf8');
+              if (observed !== expected) {
+                failedExpectations.push(
+                  `cp '${observedPath}' '${expectedPath}'`
+                );
+              }
+            } catch (error) {
+              console.log(`EE/3: ${JSON.stringify(error)}`);
+              if (error.code === 'ENOENT') {
+                failedExpectations.push(
+                  `cp '${observedPath}' '${expectedPath}'`
+                );
+                failedExpectations.push(`git add '${expectedPath}'`);
```

Comment:
What is the notebook github integration like?

Reply:
Mostly trying to delegate that to the desktop tools.

Adding local directory access next.